core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -542,7 +543,7 @@ public CompletableFuture<Optional<Throwable>> acknowledge(String memberId, List<
                     if (inFlightBatch.batchState() != RecordState.ACQUIRED) {
                         log.debug("The batch is not in the acquired state: {} for share partition: {}-{}",
                             inFlightBatch, groupId, topicIdPartition);
-                        throwable = new InvalidRequestException("The batch cannot be acknowledged. The batch is not in the acquired state.");
+                        throwable = new InvalidRecordStateException("The batch cannot be acknowledged. The batch is not in the acquired state.");
```

Comment:
Shouldn't we return same exception in other places where we see issue updating the state? i.e. there can be state per offsets as well hence we should maintain consistency.

Reply:
yes, I missed adding it for invalid record states when subset has an issue. I have added it now. Thanks!!